tests/nosetests/pyanaconda_tests/module_subscription_tests.py

```diff
@@ -798,6 +798,57 @@ def set_subscription_request_sensitive_data_keep_test(self):
         self.assertEqual(internal_request.server_proxy_password.value,
                          "foo_proxy_password")
 
+    def attached_subscription_defaults_test(self):
+        """Test the AttachedSubscription DBus structure defaults."""
+
+        # create empty AttachedSubscription structure
+        empty_request = AttachedSubscription()
+
+        # compare with expected default values
+        expected_default_dict = {
+            "name": "",
+            "service-level": "",
+            "sku": "",
+            "contract": "",
+            "start-date": "",
+            "end-date": "",
+            "consumed-entitlement-count": ""
+        }
+        # compare the empty structure with expected default values
+        self.assertEqual(
+            get_native(AttachedSubscription.to_structure(empty_request)),
+            expected_default_dict
+        )
+
+    def attached_subscription_full_test(self):
+        """Test the AttachedSubscription DBus structure that is fully populated."""
+
+        # create empty AttachedSubscription structure
+        full_request = AttachedSubscription()
+        full_request.name = "Foo Bar Beta"
+        full_request.service_level = "really good"
+        full_request.sku = "ABC123456"
+        full_request.contract = "987654321"
+        full_request.start_date = "1 January 1970"
+        full_request.end_date = "19 January 2038"
+        full_request.consumed_entitlement_count = "9001"
+
+        # compare with expected values
+        expected_default_dict = {
+            "name": "Foo Bar Beta",
+            "service-level": "really good",
+            "sku": "ABC123456",
+            "contract": "987654321",
+            "start-date": "1 January 1970",
+            "end-date": "19 January 2038",
+            "consumed-entitlement-count": "9001"
+        }
+        # compare the full structure with expected values
+        self.assertEqual(
+            get_native(AttachedSubscription.to_structure(full_request)),
```

Comment:
It would be good to have one test with variants so we can verify the types.

Reply:
Good point, switched the tests to variants. :)